python/taichi/lang/matrix.py

```diff
@@ -800,19 +772,71 @@ def identity(dt, n):
                        for i in range(n)])
 
     @staticmethod
-    @taichi_scope
     def rotation2d(alpha):
         import taichi as ti
         return Matrix([[ti.cos(alpha), -ti.sin(alpha)],
                        [ti.sin(alpha), ti.cos(alpha)]])
 
     @classmethod
     @python_scope
-    def field(cls, n, m, dtype, *args, **kwargs):
+    def field(cls,
+              n=1,
+              m=1,
+              dtype=None,
+              shape=None,
+              offset=None,
+              empty=False,
+              layout=None,
+              needs_grad=False):
         '''ti.Matrix.field'''
-        _taichi_skip_traceback = 1
-        # TODO: deprecate ad-hoc use ti.Matrix() as global (#1500:2.2/2)
-        return cls(n, m, dtype, *args, **kwargs)
+        self = cls.empty(n, m)
+        self.entries = []
+        self.n = n
+        self.m = m
+        self.dt = dtype
+        assert not impl.inside_kernel()
```

Comment:
nit: might be a bit better to move this line up so it stops eayly when assertion fails?

Reply:
In fact, this can be removed since we are already using the decorator `@python_scope`, which do an assertion for us.
```suggestion
```